Library/Homebrew/utils/curl.rb

```diff
@@ -219,12 +268,12 @@ def curl_head(*args, **options)
           # If we didn't get a `Content-Disposition` header yet, retry using `GET`.
           next if request_args.empty? &&
                   parsed_output.fetch(:responses).none? { |r| r.fetch(:headers).key?("content-disposition") }
-
-          return parsed_output if result.success?
+          break if result.success?
         end
 
         result.assert_success!
       end
+      parsed_output
```

Comment:
@MikeMcQuaid Perhaps it would be better to change the return type to `T.nilable` and return `nil` when both requests fail?

Reply:
Please pull this into a separate PR please.